### ISSUE TITLE
Reduce verbosity of the default output

### ISSUE THREAD

**Issue by vlsi** [robstoll/atrium#1333]
*Platform* (all, jvm, js): all
*Extension* (none, kotlin 1.3): none

Currently, Atrium prints the class name and the hashcode for every (?) output.

It would be nice, if the class name was printed only in case it is really needed.
For instance,
* If the classes differ, add them
* Do not print hashcode by default. Frankly speaking, it is not clear what is the purpose of hashcode in the output
* If the objects are not equal, but they refer to the same object instance, it might be helpful to print "by the way, those are the same objects, yet `.equals` produced `false`

For now, I resort to `withComponent(ObjectFormatter::class) { c -> MyCustomFormatter(...) }`, however, I incline it is worth changing the defaults for less verbose output.

In other words, instead of

```
I expected subject: [1, 2, 2, 4]        (java.util.Arrays.ArrayList <1234789>)
◆ to contain, in any order: 
  ⚬ an element which needs: 
      » to be less than: 0        (kotlin.Int <1234789>)
      » but no such element was found
  ⚬ an element which needs: 
      » to be greater than: 2        (kotlin.Int <1234789>)
      » to be less than: 4        (kotlin.Int <1234789>)
      » but no such element was found
```

it would be better to have

```
I expected subject: [1, 2, 2, 4]
◆ to contain, in any order: 
  ⚬ an element which needs: 
      » to be less than: 0
      » but no such element was found
  ⚬ an element which needs: 
      » to be greater than: 2
      » to be less than: 4
      » but no such element was found
```

**Comment by robstoll** [comment]
Thanks for the feedback, I already planned this:
- removing type and identity: https://github.com/robstoll/atrium/issues/295#issuecomment-790963793

identity hash was introduced due to an edge case I ran into more often at the time when I added the feature (non internalised strings) but I realised as well that it bothers more than it helps. 

However, I don't find the time to implement the above issue at the moment. Thus, I keep this issue and see if we can already implement this earlier.
